src/3_branch.js

```diff
@@ -149,6 +150,19 @@ Branch.prototype._referringLink = function() {
 	else { return null; }
 };
 
+/***
+ * @function Branch._publishEvent
+ * @param {string} event
+ * @param {string} subject
+ */
+Branch.prototype._publishEvent = function(event, subject) {
```

Comment:
Yeah, we could just only publish events. Then they could add listeners like:
```
branch.addListener(fn); // listen for all events

branch.addListener(fn, "willCloseBanner"); // just listen for willCloseBanner events

branch.addListener(fn, ["willCloseBanner", didCloseBanner"]); // listen for 2 or more event types
```

People will probably want to listen for specific events anyway, then we can remove the concept of subjects. Sound good?